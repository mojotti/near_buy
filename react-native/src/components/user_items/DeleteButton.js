import React, { Component } from 'react';
import { Alert, Text, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';

import { styles } from '../../static/styles/UserItemDetailsStyles';
import { DELETION_ERROR, localhost } from '../../static/constants';
import { getBearerHeaders } from '../../networking/networking';

export default class DeleteButton extends Component {
  _navigateBack = () => {
    const { goBack } = this.props.navigation;
    goBack();
  };

  _removeItemAndNavigateBack = () => {
    const url = `http://${localhost}:5000/api/v1.0/user/items/${this.props.id}`;
    fetch(url, {
      method: 'DELETE',
      headers: getBearerHeaders(this.props.token),
    })
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.ok) {
          this.props.fetchItems();
          this._navigateBack();
        }
      })
      .catch(error => {
        Alert.alert(...DELETION_ERROR);
        console.error(error);
      });
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this._removeItemAndNavigateBack}
        style={styles.deleteButton}
      >
        <Text style={styles.buttonText}>Delete item</Text>
      </TouchableHighlight>
    );
  }
}

DeleteButton.propTypes = {
  id: PropTypes.number.isRequired,
  token: PropTypes.string.isRequired,
};
