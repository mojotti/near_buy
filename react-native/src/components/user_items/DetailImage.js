import React, { Component } from 'react';
import { Dimensions, View } from 'react-native';
import ImageLoad from 'react-native-image-placeholder';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
const { height } = Dimensions.get('window');

export default class DetailImage extends Component {
  renderImage = () => {
    return (
      <View style={{ flex: 1 }}>
        <ImageLoad
          style={{ flex: 1 }}
          placeholderStyle={{
            flex: 1,
            alignSelf: 'center',
            alignItems: 'center',
            justifyContent: 'center',
          }}
          source={{ uri: this.props.url }}
          isShowActivity={false}
        />
      </View>
    );
  };

  renderPlaceholder = () => {
    return (
      <View
        style={{
          flex: 1,
          alignSelf: 'center',
          alignItems: 'center',
          justifyContent: 'center',
          paddingBottom: 40,
          backgroundColor: '#FFFFFF',
        }}
      >
        <MaterialIcons name="add-a-photo" size={height * 0.4} color="gray" />
      </View>
    );
  };

  render() {
    return this.props.url ? this.renderImage() : this.renderPlaceholder();
  }
}
