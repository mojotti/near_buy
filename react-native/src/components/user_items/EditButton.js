import React, { Component } from 'react';
import { Alert, Text, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';

import { styles } from '../../static/styles/UserItemDetailsStyles';
import { DELETION_ERROR, localhost } from '../../static/constants';
import { getHeadersForRegistering } from '../../networking/networking';

export default class EditButton extends Component {
  _getHeaders = () => {
    const headers = new Headers();
    headers.append('Authorization', `Bearer ${this.props.token}`);
    headers.append('Content-Type', 'application/json');
    return headers;
  };

  _navigateBack = () => {
    const { goBack } = this.props.navigation;
    goBack();
  };

  _updateItemAndNavigateBack = () => {
    const url = `http://${localhost}:5000/api/v1.0/items/${this.props.id}`;
    console.log(this.props.location);
    fetch(url, {
      method: 'PUT',
      headers: this._getHeaders(),
      body: JSON.stringify({
        id: this.props.item.id,
        title: this.props.item.title,
        description: this.props.item.description,
        price: this.props.item.price,
        longitude: this.props.location.longitude,
        latitude: this.props.location.latitude,
        sold: false,
      }),
      N,
    })
      .then(response => response.json())
      .then(responseJson => {
        console.log('response', responseJson);
        this.props.fetchItems();
        this._navigateBack();
      })
      .catch(error => {
        Alert.alert(...DELETION_ERROR);
        console.error(error);
      });
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this._updateItemAndNavigateBack}
        style={styles.editButton}
      >
        <Text style={styles.buttonText}>Edit item</Text>
      </TouchableHighlight>
    );
  }
}

EditButton.propTypes = {
  id: PropTypes.number.isRequired,
  token: PropTypes.string.isRequired,
};
