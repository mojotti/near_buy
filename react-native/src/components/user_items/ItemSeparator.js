import React, { Component } from 'react';
import { View } from 'react-native';
import { styles } from '../../static/styles/ListViewItemStyle';

export default class ItemSeparator extends Component {
  render() {
    return <View style={styles.separator} />;
  }
}
