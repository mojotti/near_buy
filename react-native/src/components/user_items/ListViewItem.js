import React from 'react';
import { Text, View } from 'react-native';
import { styles } from '../../static/styles/ListViewItemStyle';

export class ListViewItem extends React.Component {
  render() {
    return (
      <View style={[styles.item, styles.separator]}>
        <Text>{`Title: ${this.props.item.title}`}</Text>
        <Text>{`Description: ${this.props.item.description}`}</Text>
        <Text>{`Price: ${this.props.item.price} â‚¬`}</Text>
      </View>
    );
  }
}
