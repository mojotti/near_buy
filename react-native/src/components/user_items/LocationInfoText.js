import React from 'react';
import { Text, View } from 'react-native';
import { styles } from '../../static/styles/UserItemDetailsStyles';

export default class LocationInfoText extends React.Component {
  render() {
    return (
      <View style={{ alignSelf: 'stretch' }}>
        <Text
          style={styles.infoText}
        >{`\u2022 Your exact location is not visible to others`}</Text>
        <Text style={styles.infoText}>
          {`\u2022`}
          <Text style={{ color: 'blue' }}> Blue</Text>
          <Text>{` color indicates your item's location`}</Text>
        </Text>
        <Text style={styles.infoText}>
          {`\u2022`}
          <Text style={{ color: 'red' }}> Red</Text>
          <Text>{` color indicates your location`}</Text>
        </Text>
      </View>
    );
  }
}
