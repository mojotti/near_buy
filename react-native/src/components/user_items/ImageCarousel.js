import React from 'react';
import PropTypes from 'prop-types';
import Carousel from 'react-native-looped-carousel';
import { styles } from '../../static/styles/UserItemDetailsStyles';
import DetailImage from './DetailImage';

export default class ImageCarousel extends React.Component {
  render() {
    return (
      <Carousel
        style={styles.carousel}
        autoplay={false}
        pageInfo
        currentPage={0}
        pageInfoBackgroundColor={'#292929'}
        pageInfoTextStyle={{ color: '#FFFFFF' }}
    // arrows are disabled for now since they don't work as intended on ios
        arrows={false}
        arrowStyle={styles.arrowStyle}
        leftArrowText={' '}
        rightArrowText={' '}
      >
        <DetailImage
          url={this.props.imageUrls[0]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[1]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[2]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
        <DetailImage
          url={this.props.imageUrls[3]}
          id={this.props.id}
          numOfPics={this.props.numOfPics}
          onImageUpload={this.props.onImageUpload}
        />
      </Carousel>
    );
  }
}

ImageCarousel.propTypes = {
  imageUrls: PropTypes.arrayOf(PropTypes.string).isRequired,
  id: PropTypes.number.isRequired,
  onImageUpload: PropTypes.func.isRequired,
  numOfPics: PropTypes.number.isRequired,
};
