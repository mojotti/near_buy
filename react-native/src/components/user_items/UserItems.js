'use strict';

import React from 'react';
import {
  ActivityIndicator,
  FlatList,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { localhost, userHasNoItemsText } from '../../static/constants';
import UserItem from './UserItem';
import { styles } from '../../static/styles/ItemsStyles';
import ItemSeparator from './ItemSeparator';
import { getBearerHeaders } from '../../networking/networking';

export class UserItems extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {},
      isLoaded: false,
      isRefreshing: false,
    };
  }

  static navigationOptions = ({ navigation }) => {
    const { navigate } = navigation;
    return {
      title: 'My items',
      headerLeft: (
        <View style={{ paddingHorizontal: 10 }}>
          <TouchableOpacity onPress={() => navigation.navigate('DrawerOpen')}>
            <Icon name="menu" size={30} color="#4d4dff" />
          </TouchableOpacity>
        </View>
      ),
      headerRight: (
        <Text onPress={() => navigate('NewItem')} style={styles.headerButton}>
          New item
        </Text>
      ),
    };
  };

  componentDidMount() {
    this.fetchData();
  }

  fetchData = () => {
    const url = `http://${localhost}:5000/api/v1.0/user/items`;
    this.setState(() => ({ isRefreshing: true }));

    fetch(url, {
      method: 'GET',
      headers: getBearerHeaders(this.props.token),
    })
      .then(response => response.json())
      .then(responseJson => {
        this.handleAllItemsResponse(responseJson);
      })
      .catch(error => {
        console.error(error);
        this.setState(() => ({ isRefreshing: false }));
      });
  };

  handleAllItemsResponse(responseJson) {
    console.log('Items', responseJson.items);
    this.setState(() => ({
      isLoaded: true,
      data: responseJson.items,
      isRefreshing: false,
    }));
  }

  renderUserData() {
    if (!this.state.isLoaded) {
      return (
        <ActivityIndicator
          size="large"
          color="#0000ff"
          style={styles.activityIndicator}
        />
      );
    }
    if (!this.state.data || this.state.data === 'no items') {
      return <Text style={[styles.infoText]}>{userHasNoItemsText}</Text>;
    } else {
      return (
        <FlatList
          data={this.state.data}
          renderItem={rowData => {
            return (
              <UserItem
                item={rowData.item}
                itemId={rowData.index}
                fetchItems={this.fetchData}
                navigation={this.props.navigation}
              />
            );
          }}
          keyExtractor={(item, index) => index.toString()}
          onRefresh={this.fetchData}
          refreshing={this.state.isRefreshing}
          ItemSeparatorComponent={ItemSeparator}
        />
      );
    }
  }

  render() {
    return <View style={[styles.container]}>{this.renderUserData()}</View>;
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    username: state.authorizationReducer.username,
    token: state.authorizationReducer.token,
  };
};

export default connect(mapStateToProps, null)(UserItems);
