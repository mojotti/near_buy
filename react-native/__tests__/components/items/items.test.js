import React from 'react';
import 'isomorphic-fetch'; // for headers, fetch, etc.
import renderer from 'react-test-renderer';

import { UserItems } from '../../../src/components/user_items/UserItems';
import { sample } from '../../../src/static/samples/ItemSample';

describe('UserItems', () => {
  it('render correctly when response has user_items', () => {
    fetch.mockResponseSuccess(sample);

    const items = renderer.create(<UserItems />);
    items.getInstance().handleAllItemsRequest(sample);

    expect(items.toJSON()).toMatchSnapshot();
  });
});
