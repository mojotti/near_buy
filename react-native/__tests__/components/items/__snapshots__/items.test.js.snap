// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItems render correctly when response has user_items 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 6,
      }
    }
    renderRow={[Function]}
    renderScrollComponent={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "azure",
              "padding": 10,
              "width": 750,
            },
            Object {
              "borderBottomColor": "gainsboro",
              "borderBottomWidth": 1,
              "marginBottom": 1,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Title: Item 1
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Description: Description. Very awesome item.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Price: 50 €
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "azure",
              "padding": 10,
              "width": 750,
            },
            Object {
              "borderBottomColor": "gainsboro",
              "borderBottomWidth": 1,
              "marginBottom": 1,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Title: New item 500
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Description: Description 600
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Price: 120 €
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "azure",
              "padding": 10,
              "width": 750,
            },
            Object {
              "borderBottomColor": "gainsboro",
              "borderBottomWidth": 1,
              "marginBottom": 1,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Title: Jee
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Description: LOl
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Price: 200 €
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "azure",
              "padding": 10,
              "width": 750,
            },
            Object {
              "borderBottomColor": "gainsboro",
              "borderBottomWidth": 1,
              "marginBottom": 1,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Title: JEees Jeees
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Description: Jees
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Price: 100 €
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "azure",
              "padding": 10,
              "width": 750,
            },
            Object {
              "borderBottomColor": "gainsboro",
              "borderBottomWidth": 1,
              "marginBottom": 1,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Title: Neoa
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Description: Otoaeeontaonaoooenotaontaoenoteaoneoanoao
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Price: 10200 €
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "azure",
              "padding": 10,
              "width": 750,
            },
            Object {
              "borderBottomColor": "gainsboro",
              "borderBottomWidth": 1,
              "marginBottom": 1,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Title: Jees
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Description: JEeeeeabaeoaneo
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Price: 100 €
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;
